src/System.Linq/src/System/Linq/OrderedEnumerable.cs

```diff
@@ -102,6 +109,31 @@ internal TElement[] ToArray(int minIdx, int maxIdx)
             return array;
         }
 
+        internal List<TElement> ToList(int minIdx, int maxIdx)
+        {
+            Buffer<TElement> buffer = new Buffer<TElement>(source);
+            int count = buffer.count;
+            if (count <= minIdx) return new List<TElement>();
+            if (count <= maxIdx) maxIdx = count - 1;
+            if (minIdx == maxIdx) return new List<TElement>(1) { GetEnumerableSorter().ElementAt(buffer.items, count, minIdx) };
+            int[] map = SortedMap(buffer, minIdx, maxIdx);
+            List<TElement> list = new List<TElement>(maxIdx - minIdx + 1);
+            while (minIdx <= maxIdx)
+            {
+                list.Add(buffer.items[map[minIdx]]);
+                ++minIdx;
+            }
+            return list;
+        }
+
+        internal int GetCount(int minIdx, int maxIdx, bool onlyIfCheap)
+        {
+            int count = GetCount(onlyIfCheap);
+            if (count <= 0) return count;
+            if (count <= minIdx) return 0;
```

Comment:
I think "onlyIfCheap" is unnecessary. Statically you can only know "onlyIfPossible" - ie if we have something list-like with finate count we can get that, otherwise if we have just some, potentially infinitely long sequence, we cannot produce count. 
And having "onlyIfPossible" flag would be kind of silly since it would be always passed as true.

Basically, I am not sure I see a case where the "onlyIfCheap" will ever be passed false.